server/proxy/channels/pf_channel_drdynvc.c

```diff
@@ -409,6 +439,7 @@ static PfChannelResult DynvcTrackerPeekFn(ChannelStateTracker* tracker, BOOL fir
 			         direction);
 			channelTracker_setMode(tracker, CHANNEL_TRACKER_PASS);
 			dynChannel->openStatus = CHANNEL_OPENSTATE_CLOSED;
+			HashTable_Remove(dynChannelContext->channels, &dynChannel->channelId);
```

Comment:
With this the CLOSE_ACK (CLOSE response) will be dropped by the proxy. Shall we remove the channel only when we receive the CLOSE ack, or after a timeout ?

Reply:
@hardening changed the behaviour:
1. `HashTable_Remove` is now not called on close or open failure (the tracker state remains in the list)
2. If a channel open encounters a already allocated state tracker, a message is printed and the previous state removed

this way we `CLOSE_ACK` will work and the overhead is just some allocated memory until we reuse the channel.